--- src/Terraria\Terraria.Achievements\AchievementManager.cs
+++ src/TerraCustom\Terraria.Achievements\AchievementManager.cs
@@ -20,11 +20,17 @@
 		}
 
 		private string _savePath;
+
 		private bool _isCloudSave;
+
 		private Dictionary<string, Achievement> _achievements = new Dictionary<string, Achievement>();
+
 		private readonly JsonSerializerSettings _serializerSettings = new JsonSerializerSettings();
+
 		private byte[] _cryptoKey;
+
 		private Dictionary<string, int> _achievementIconIndexes = new Dictionary<string, int>();
+
 		private static object _ioLock = new object();
 
 		public event Achievement.AchievementCompleted OnAchievementCompleted;

