--- src/Terraria\Terraria.Achievements\Achievement.cs
+++ src/TerraCustom\Terraria.Achievements\Achievement.cs
@@ -12,14 +12,22 @@
 		public delegate void AchievementCompleted(Achievement achievement);
 
 		private static int _totalAchievements;
+
 		public readonly string Name;
+
 		public readonly string FriendlyName;
+
 		public readonly string Description;
+
 		public readonly int Id = Achievement._totalAchievements++;
+
 		private AchievementCategory _category;
+
 		private IAchievementTracker _tracker;
+
 		[JsonProperty("Conditions")]
 		private Dictionary<string, AchievementCondition> _conditions = new Dictionary<string, AchievementCondition>();
+
 		private int _completedCount;
 
 		public event Achievement.AchievementCompleted OnCompleted;

