--- src/tModLoader\Terraria.Utilities\PlatformUtilities.cs
+++ src/TerraCustom\Terraria.Utilities\PlatformUtilities.cs
@@ -87,8 +_,7 @@
 		{
 #if WINDOWS
 			string path = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), "My Games");
-			path = Path.Combine(path, "Terraria");
-			return Path.Combine(path, "ModLoader");
+			return Path.Combine(path, "Terraria");
 #endif
 #if MAC
 			string text = Environment.GetEnvironmentVariable("HOME");
@@ -97,8 +_,7 @@
 				return ".";
 			}
 			text += "/Library/Application Support";
-			text = Path.Combine(text, "Terraria");
-			return Path.Combine(text, "ModLoader");
+			return Path.Combine(text, "Terraria");
 #endif
 #if LINUX
 			string text = Environment.GetEnvironmentVariable("XDG_DATA_HOME");
@@ -111,8 +_,7 @@
 				}
 				text += "/.local/share";
  			}
- 			text = Path.Combine(text, "Terraria");
-			return Path.Combine(text, "ModLoader");
+ 			return Path.Combine(text, "Terraria");
 #endif
 		}
 		#if !WINDOWS

