--- src/tModLoader\Terraria.csproj
+++ src/TerraCustom\Terraria.csproj
@@ -13,7 +_,7 @@
     <LangVersion>7.3</LangVersion>
     <PlatformTarget>AnyCPU</PlatformTarget>
     <Optimize>true</Optimize>
-    <OutputName>tModLoader</OutputName>
+    <OutputName>TerraCustom</OutputName>
   </PropertyGroup>
   <PropertyGroup Condition="$(Configuration.Contains('Server'))">
     <OutputType>Exe</OutputType>
