--- src/tModLoader\Terraria.IO\WorldFile.cs
+++ src/TerraCustom\Terraria.IO\WorldFile.cs
@@ -341,9 +_,16 @@
 				if (text != null && array2 != null)
 					FileUtilities.WriteAllBytes(text, array2, useCloudSaving);
 
-				WorldIO.Save(Main.worldPathName, useCloudSaving);
+				if (Main.tModLoaderModsLoaded)
+				{
+					WorldIO.Save(Main.worldPathName, useCloudSaving);
+				}
 				WorldGen.saveLock = false;
+			}
+			if (Main.setting.generateLeveledRPGSave && !Main.tModLoaderModsLoaded)
+			{
+				LeveledRPGMod.LeveledRPGModUtilities.saveEXP(useCloudSaving, Main.worldPathName);
 			}
 
 			Main.serverGenLock = false;
 		}
