--- src/tModLoader\Terraria.IO\WorldFile.cs
+++ src/TerraCustom\Terraria.IO\WorldFile.cs
@@ -400,8 +_,15 @@
 				{
 					FileUtilities.WriteAllBytes(text, array2, useCloudSaving);
 				}
-				WorldIO.WriteModFile(Main.worldPathName, useCloudSaving);
+				if (Main.tModLoaderModsLoaded)
+				{
+					WorldIO.WriteModFile(Main.worldPathName, useCloudSaving);
+				}
 				WorldGen.saveLock = false;
+			}
+			if (Main.setting.generateLeveledRPGSave && !Main.tModLoaderModsLoaded)
+			{
+				LeveledRPGMod.LeveledRPGModUtilities.saveEXP(useCloudSaving, Main.worldPathName);
 			}
 			Main.serverGenLock = false;
 		}

