--- src/tModLoader\Terraria.GameContent.Biomes\EnchantedSwordBiome.cs
+++ src/TerraCustom\Terraria.GameContent.Biomes\EnchantedSwordBiome.cs
@@ -43,5 +_,5 @@
 			WorldUtils.Gen(new Point(origin.X, result.Y + 10), new Shapes.Rectangle(1, origin.Y - result.Y - 9), Actions.Chain(new Modifiers.Blotches(2, 0.2), new Actions.ClearTile().Output(data), new Modifiers.Expand(1), new Modifiers.OnlyTiles(53), new Actions.SetTile(397).Output(data)));
 			WorldUtils.Gen(new Point(origin.X, result.Y + 10), new ModShapes.All(data), new Actions.SetFrames(frameNeighbors: true));
-			if (GenBase._random.Next(3) == 0)
+			if (GenBase._random.Next(3) == 0 || Main.setting.ForceEnchantedSwordShrineReal)
 				WorldGen.PlaceTile(point2.X, point2.Y - 15, 187, mute: true, forced: false, -1, 17);
 			else
