--- src/Terraria\Terraria\Lang.cs
+++ src/TerraCustom\Terraria\Lang.cs
@@ -35599,6 +_,7 @@
 					Lang.prefix[k] = string.Empty;
 				}
 			}
+			/*
 			for (int l = 0; l < Main.recipe.Length; l++)
 			{
 				if (Main.recipe[l].createItem.name != null && Main.recipe[l].createItem.name != "" && Main.recipe[l].createItem.netID != 0)
@@ -35612,6 +_,7 @@
 					}
 				}
 			}
+			*/
 		}
 
 		public static string deathMsg(int plr = -1, int npc = -1, int proj = -1, int other = -1)

