--- src/tModLoader\Terraria\NPC.cs
+++ src/TerraCustom\Terraria\NPC.cs
@@ -62147,15 +_,15 @@
 			}
 			if (NPCLoader.StrikeNPC(this, ref num, num2, ref knockBack, hitDirection, ref crit))
 			{
-				num = Main.CalculateDamage((int)num, num2);
-				if (crit)
-				{
-					num *= 2.0;
-				}
-				if (this.takenDamageMultiplier > 1f)
-				{
-					num *= (double)this.takenDamageMultiplier;
-				}
+			num = Main.CalculateDamage((int)num, num2);
+			if (crit)
+			{
+				num *= 2.0;
+			}
+			if (this.takenDamageMultiplier > 1f)
+			{
+				num *= (double)this.takenDamageMultiplier;
+			}
 			}
 			if ((this.takenDamageMultiplier > 1f || Damage != 9999) && this.lifeMax > 1)
 			{

