--- src/tModLoader\Terraria\Program.cs
+++ src/TerraCustom\Terraria\Program.cs
@@ -10,7 +_,7 @@
 #endif
 using Terraria.Initializers;
 using Terraria.ModLoader;
-using Terraria.Social;
+//using Terraria.Social;
 
 namespace Terraria
 {
@@ -112,7 +_,7 @@
 			{
 				try
 				{
-					SocialAPI.Initialize(null);
+					//SocialAPI.Initialize(null);
 					LaunchInitializer.LoadParameters(main);
 					Main.OnEngineLoad += delegate
 					{

