--- src/tModLoader\Terraria\Program.cs
+++ src/TerraCustom\Terraria\Program.cs
@@ -235,11 +_,11 @@
 			// These need to be here since LoadParameters is too late since new Main() and its static initializers happens first.
 			if (Program.LaunchParameters.ContainsKey("-savedirectory"))
 			{
-				Program.LaunchParameters["-savedirectory"] = Path.Combine(Program.LaunchParameters["-savedirectory"], "ModLoader");
+				//Program.LaunchParameters["-savedirectory"] = Path.Combine(Program.LaunchParameters["-savedirectory"], "ModLoader");
 			}
 			if (Program.LaunchParameters.ContainsKey("-tmlsavedirectory"))
 			{
-				Program.LaunchParameters["-savedirectory"] = Program.LaunchParameters["-tmlsavedirectory"];
+				//Program.LaunchParameters["-savedirectory"] = Program.LaunchParameters["-tmlsavedirectory"];
 			}
 			ErrorLogger.ClearLogs();
 			ThreadPool.SetMinThreads(8, 8);

