--- src/tModLoader\Terraria\Program.cs
+++ src/TerraCustom\Terraria\Program.cs
@@ -19,7 +_,7 @@
 using Terraria.Initializers;
 using Terraria.Localization;
 using Terraria.ModLoader;
-using Terraria.Social;
+//using Terraria.Social;
 using Terraria.Utilities;
 
 namespace Terraria
@@ -250,7 +_,7 @@
 				{
 					Program.InitializeConsoleOutput();
 					Lang.InitializeLegacyLocalization();
-					SocialAPI.Initialize(null);
+					//SocialAPI.Initialize(null);
 					LaunchInitializer.LoadParameters(main);
 					Main.OnEnginePreload += new Action(Program.StartForceLoad);
 #if CLIENT

