--- src/Terraria\Terraria\Program.cs
+++ src/TerraCustom\Terraria\Program.cs
@@ -9,8 +9,8 @@
 using System.Windows.Forms;
 #endif
 using Terraria.Initializers;
-using Terraria.Social;
 
+//using Terraria.Social;
 namespace Terraria
 {
 	public static class Program
@@ -102,7 +102,7 @@
 			{
 				try
 				{
-					SocialAPI.Initialize(null);
+					//SocialAPI.Initialize(null);
 					LaunchInitializer.LoadParameters(main);
 					Main.OnEngineLoad += delegate
 					{

