--- src/Terraria\Terraria\Recipe.cs
+++ src/TerraCustom\Terraria\Recipe.cs
@@ -286,7 +_,7 @@
 		}
 
 		public static void FindRecipes()
-		{
+		{/*
 			int num = Main.availableRecipe[Main.focusRecipe];
 			float num2 = Main.availableRecipeY[Main.focusRecipe];
 			for (int i = 0; i < Recipe.maxRecipes; i++)
@@ -456,10 +_,11 @@
 			{
 				Main.availableRecipeY[num8] -= num7;
 			}
+		*/
 		}
 
 		public static void SetupRecipes()
-		{
+		{/*
 			int stack = 5;
 			int stack2 = 2;
 			Recipe.newRecipe.createItem.SetDefaults(8, false);
@@ -10139,10 +_,11 @@
 				Main.recipe[m].createItem.checkMat();
 			}
 			int arg_27EA8_0 = Recipe.numRecipes;
+		*/
 		}
 
 		private static void PlatformReturn()
-		{
+		{/*
 			int num = Recipe.numRecipes;
 			for (int i = 0; i < num; i++)
 			{
@@ -10165,10 +_,11 @@
 					Main.recipe[i + 1] = recipe;
 				}
 			}
+		*/
 		}
 
 		private static void WallReturn()
-		{
+		{/*
 			int num = Recipe.numRecipes;
 			for (int i = 0; i < num; i++)
 			{
@@ -10191,10 +_,11 @@
 					Main.recipe[i + 1] = recipe;
 				}
 			}
+		*/
 		}
 
 		private static void AddRecipe()
-		{
+		{/*
 			if (Recipe.newRecipe.requiredTile[0] == 13)
 			{
 				Recipe.newRecipe.alchemy = true;
@@ -10202,6 +_,7 @@
 			Main.recipe[Recipe.numRecipes] = Recipe.newRecipe;
 			Recipe.newRecipe = new Recipe();
 			Recipe.numRecipes++;
+		*/
 		}
 	}
 }

