--- src/tModLoader\Terraria\Recipe.cs
+++ src/TerraCustom\Terraria\Recipe.cs
@@ -143,15 +_,15 @@
 		}
 
 		public bool ItemMatches(Item invItem, Item reqItem)
-		{
+			{
 			return invItem.IsTheSameAs(reqItem) || this.useWood(invItem.type, reqItem.type)
 			|| this.useIronBar(invItem.type, reqItem.type) || this.useSand(invItem.type, reqItem.type)
 			|| this.useFragment(invItem.type, reqItem.type) || this.usePressurePlate(invItem.type, reqItem.type)
 			|| (this is ModRecipe && ((ModRecipe)this).UseCraftGroup(invItem.type, reqItem.type));
-		}
+			}
 
 		public bool useWood(int invType, int reqType)
-		{
+			{
 			CraftGroup group = CraftGroup.Wood;
 			return this.anyWood && group.Items.Contains(invType) && group.Items.Contains(reqType);
 		}

