--- src/tModLoader\Terraria\Tile.cs
+++ src/TerraCustom\Terraria\Tile.cs
@@ -28,6 +_,7 @@
 		public byte bTileHeader3;
 		public short frameX;
 		public short frameY;
+		internal static int WireTypeOverride = -1;
 
 		public int collisionType
 		{
@@ -458,6 +_,18 @@
 
 		public void wire(bool wire)
 		{
+			if(WireTypeOverride > 0) // -1 is rand, 0 is 0
+			{
+				if (WireTypeOverride == 1)
+				{
+					wire2(wire);
+				}
+				else
+				{
+					wire3(wire);
+				}
+				return;
+			}
 			if (wire)
 			{
 				this.sTileHeader |= 128;

