--- src/tModLoader\Terraria\Tile.cs
+++ src/TerraCustom\Terraria\Tile.cs
@@ -18,5 +_,6 @@
 		public byte bTileHeader3;
 		public short frameX;
 		public short frameY;
+		internal static int WireTypeOverride = -1;
 		public const int Type_Solid = 0;
 		public const int Type_Halfbrick = 1;
@@ -351,8 +_,20 @@
 		public bool wire() => (sTileHeader & 0x80) == 128;
 
 		public void wire(bool wire) {
+			if(WireTypeOverride > 0) // -1 is rand, 0 is 0
+			{
+				if (WireTypeOverride == 1)
+				{
+					wire2(wire);
+				}
+				else
+				{
+					wire3(wire);
+				}
+				return;
+			}
 			if (wire)
 				sTileHeader |= 128;
 			else
 				sTileHeader = (ushort)(sTileHeader & 0xFF7F);
 		}
