--- src/Terraria\Terraria\Tile.cs
+++ src/TerraCustom\Terraria\Tile.cs
@@ -23,6 +23,7 @@
 		public byte bTileHeader3;
 		public short frameX;
 		public short frameY;
+        public bool Worldspawned = false;
 
 		public int collisionType
 		{
@@ -366,6 +367,22 @@
 				return;
 			}
 			this.bTileHeader3 &= 239;
+		}
+
+		public void wire(bool wi, int type)
+		{
+			if (type == 0)
+			{
+				wire(wi);
+			}
+			else if (type == 1)
+			{
+				wire2(wi);
+			}
+			else
+			{
+				wire3(wi);
+			}
 		}
 
 		public bool wire()

