--- src/tModLoader\Terraria\WindowsLaunch.cs
+++ src/TerraCustom\Terraria\WindowsLaunch.cs
@@ -3,7 +_,7 @@
 using System.IO;
 using System.Reflection;
 using System.Runtime.InteropServices;
-using Terraria.Social;
+//using Terraria.Social;
 namespace Terraria
 {
 	internal static class WindowsLaunch
@@ -39,7 +_,7 @@
 			}
 			if (flag)
 			{
-				SocialAPI.Shutdown();
+				//SocialAPI.Shutdown();
 			}
 			return true;
 		}
@@ -63,12 +_,12 @@
 					result = Assembly.Load(array);
 				}
 				return result;
-            };
+			};
 #if SERVER
 			WindowsLaunch._handleRoutine = new WindowsLaunch.HandlerRoutine(WindowsLaunch.ConsoleCtrlCheck);
 			WindowsLaunch.SetConsoleCtrlHandler(WindowsLaunch._handleRoutine, true);
 #endif
-            Program.LaunchGame(args);
+			Program.LaunchGame(args);
 		}
 	}
 }

