--- src/Terraria\Terraria\WindowsLaunch.cs
+++ src/TerraCustom\Terraria\WindowsLaunch.cs
@@ -3,7 +3,7 @@
 using System.IO;
 using System.Reflection;
 using System.Runtime.InteropServices;
-using Terraria.Social;
+//using Terraria.Social;
 namespace Terraria
 {
 	internal static class WindowsLaunch
@@ -39,7 +39,7 @@
 			}
 			if (flag)
 			{
-				SocialAPI.Shutdown();
+				//SocialAPI.Shutdown();
 			}
 			return true;
 		}

