--- src/Terraria\Terraria\WorldSections.cs
+++ src/TerraCustom\Terraria\WorldSections.cs
@@ -9,10 +9,15 @@
 		private struct IterationState
 		{
 			public Vector2 centerPos;
+
 			public int X;
+
 			public int Y;
+
 			public int leg;
+
 			public int xDir;
+
 			public int yDir;
 
 			public void Reset()
@@ -27,11 +32,17 @@
 		}
 
 		private int width;
+
 		private int height;
+
 		private BitsByte[] data;
+
 		private int mapSectionsLeft;
+
 		private int frameSectionsLeft;
+
 		private WorldSections.IterationState prevFrame;
+
 		private WorldSections.IterationState prevMap;
 
 		public int FrameSectionsLeft

