--- src/tModLoader/Terraria.ModLoader.UI/Interface.cs
+++ src/TerraCustom/Terraria.ModLoader.UI/Interface.cs
@@ -378,7 +_,7 @@
 		internal static void MessageBoxShow(string text, string caption = null) {
 			// MessageBox.Show fails on Mac, this method will open a text file to show a message.
 			caption = caption ?? "Terraria: Error" + $" ({ModLoader.versionedName})";
-			string message = Language.GetTextValue("tModLoader.ClientLogHint", text, Path.Combine(Main.SavePath, "Logs"));
+			string message = Language.GetTextValue("tModLoader.ClientLogHint", text, Logging.LogDir);
 #if !MAC
 			System.Windows.Forms.MessageBox.Show(message, caption);
 #else
