--- src/tModLoader/Terraria.ModLoader.UI/Interface.cs
+++ src/TerraCustom/Terraria.ModLoader.UI/Interface.cs
@@ -378,7 +_,7 @@
 		internal static void MessageBoxShow(string text, string caption = null) {
 			// MessageBox.Show fails on Mac, this method will open a text file to show a message.
 			caption = caption ?? "Terraria: Error" + $" ({ModLoader.versionedName})";
-			string message = $"{text}\n\nA client.log file containing error information has been generated in\n{Path.Combine(Main.SavePath, "Logs")}\n(You will need to share this file if asking for help)";
+			string message = $"{text}\n\nA client.log file containing error information has been generated in\n{Logging.LogDir}\n(You will need to share this file if asking for help)";
 #if !MAC
 			System.Windows.Forms.MessageBox.Show(message, caption);
 #else
