--- src/tModLoader\Terraria.World.Generation\WorldGenerator.cs
+++ src/TerraCustom\Terraria.World.Generation\WorldGenerator.cs
@@ -6,6 +_,7 @@
 using Terraria.ModLoader;
 using Terraria.ModLoader.UI;
 using Terraria.Utilities;
+using Terraria.IO;
 
 namespace Terraria.World.Generation
 {
@@ -26,5 +_,6 @@
 
 		public void GenerateWorld(GenerationProgress progress = null) {
+			string originalWorldName = Main.worldName;
 			Stopwatch stopwatch = new Stopwatch();
 			float num = 0f;
 			foreach (GenPass pass in _passes) {
@@ -36,5 +_,6 @@
 
 			progress.TotalWeight = num;
 			Main.menuMode = 888;
+			int a = 0;
 			Main.MenuUI.SetState(new UIWorldLoad(progress));
 			foreach (GenPass pass2 in _passes) {
@@ -61,7 +_,17 @@
 				}
 				progress.End();
 				stopwatch.Reset();
+				if (Main.setting.GenerateWldEachStep)
+				{
+					a++;
+					Main.worldName = originalWorldName + "_" + a.ToString("D2");
+					Main.WorldFileMetadata = FileMetadata.FromCurrentSettings(FileType.World);
+					Main.ActiveWorldFileData = WorldFile.CreateMetadata(Main.worldName, false, Main.expertMode);
+					Terraria.IO.WorldFile.saveWorld(false, true);
+				}
 			}
+			Main.worldName = originalWorldName;
+			Main.ActiveWorldFileData = WorldFile.CreateMetadata(Main.worldName, false, Main.expertMode);
 		}
 	}
 }
