--- src/tModLoader\Terraria.World.Generation\WorldGenerator.cs
+++ src/TerraCustom\Terraria.World.Generation\WorldGenerator.cs
@@ -2,7 +_,8 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using Terraria.GameContent.UI.States;
 using Terraria.Utilities;
+using Terraria.IO;
 
 namespace Terraria.World.Generation
 {
@@ -18,6 +_,7 @@
 
 		public void GenerateWorld(GenerationProgress progress = null)
 		{
+			string originalWorldName = Main.worldName;
 			Stopwatch stopwatch = new Stopwatch();
 			float num = 0f;
 			foreach (GenPass current in this._passes)
@@ -31,6 +_,7 @@
 			progress.TotalWeight = num;
 			string text = "";
 			Main.menuMode = 888;
+			int a = 0;
 			Main.MenuUI.SetState(new UIWorldLoad(progress));
 			foreach (GenPass current2 in this._passes)
 			{
@@ -49,7 +_,17 @@
 						",\n"
 					});
 				stopwatch.Reset();
+				if (Main.setting.GenerateWldEachStep)
+				{
+					a++;
+					Main.worldName = originalWorldName + "_" + a.ToString("D2");
+					Main.WorldFileMetadata = FileMetadata.FromCurrentSettings(FileType.World);
+					Main.ActiveWorldFileData = WorldFile.CreateMetadata(Main.worldName, false, Main.expertMode);
+					Terraria.IO.WorldFile.saveWorld(false, true);
+				}
 			}
+			Main.worldName = originalWorldName;
+			Main.ActiveWorldFileData = WorldFile.CreateMetadata(Main.worldName, false, Main.expertMode);
 		}
 	}
 }

