--- src/Terraria\Terraria.Initializers\DyeInitializer.cs
+++ src/TerraCustom\Terraria.Initializers\DyeInitializer.cs
@@ -460,11 +_,13 @@
 
 		private static void FixRecipes()
 		{
+			/*
 			for (int i = 0; i < Recipe.maxRecipes; i++)
 			{
 				Main.recipe[i].createItem.dye = (byte)GameShaders.Armor.GetShaderIdFromItemId(Main.recipe[i].createItem.type);
 				Main.recipe[i].createItem.hairDye = GameShaders.Hair.GetShaderIdFromItemId(Main.recipe[i].createItem.type);
 			}
+			*/
 		}
 	}
 }

