--- src/tModLoader\Terraria.ModLoader.Config\ConfigManager.cs
+++ src/TerraCustom\Terraria.ModLoader.Config\ConfigManager.cs
@@ -49,7 +_,7 @@
 			//new ColorJsonConverter(),
 		};
 
-		public static readonly string ModConfigPath = Path.Combine(Main.SavePath, "Mod Configs");
+		public static readonly string ModConfigPath = Path.Combine(Main.TModLoaderSavePath, "Mod Configs");
 		public static readonly string ServerModConfigPath = Path.Combine(Main.SavePath, "Mod Configs", "Server");
 
 		internal static void Add(ModConfig config)

