--- src/tModLoader\Terraria.ModLoader\Logging.cs
+++ src/TerraCustom\Terraria.ModLoader\Logging.cs
@@ -22,8 +_,8 @@
 {
 	public static class Logging
 	{
-		public static readonly string LogDir = Path.Combine(Program.SavePath, "Logs");
+		public static readonly string LogDir = Path.Combine(Program.TerraCustomSavePath, "Logs");
 		public static readonly string LogArchiveDir = Path.Combine(LogDir, "Old");
 		public static string LogPath { get; private set; }
 
 		internal static ILog Terraria { get; } = LogManager.GetLogger("Terraria");
