--- src/tModLoader\Terraria.ModLoader\ModLoader.cs
+++ src/TerraCustom\Terraria.ModLoader\ModLoader.cs
@@ -135,7 +_,24 @@
 					OnSuccessfulLoad = null;
 				}
 				else {
-					Main.menuMode = 0;
+					//Main.menuMode = 0;
+					if (!Main.firstModLoad)
+					{
+						Main.menuMode = (int)TerraCustom.MenuModes.Settings;
+					}
+					else
+					{
+						Main.menuMode = (int)TerraCustom.MenuModes.ChooseWorldSize;
+					}
+					if (Mods.Length > 1)
+					{
+						Main.tModLoaderModsLoaded = true;
+					}
+					else
+					{
+						Main.tModLoaderModsLoaded = false;
+					}
+					Main.firstModLoad = false;
 				}
 			}
 			catch when (token.IsCancellationRequested) {
@@ -339,8 +_,10 @@
 			if (LastLaunchedTModLoaderVersion < version)
 				ShowWhatsNew = true;
 			*/
+			/* Disable Welcome message since it is reading from vanilla config file.
 			if (LastLaunchedTModLoaderVersion == new Version(0, 0))
 				ShowFirstLaunchWelcomeMessage = true;
+			*/
 		}
 
 		/// <summary>

