--- src/tModLoader/Terraria.ModLoader/ModLoader.cs
+++ src/TerraCustom/Terraria.ModLoader/ModLoader.cs
@@ -141,7 +_,24 @@
 					OnSuccessfulLoad();
 				}
 				else {
-					Main.menuMode = 0;
+					//Main.menuMode = 0;
+					if (!Main.firstModLoad)
+					{
+						Main.menuMode = (int)TerraCustom.MenuModes.Settings;
+					}
+					else
+					{
+						Main.menuMode = (int)TerraCustom.MenuModes.ChooseWorldSize;
+					}
+					if (Mods.Length > 1)
+					{
+						Main.tModLoaderModsLoaded = true;
+					}
+					else
+					{
+						Main.tModLoaderModsLoaded = false;
+					}
+					Main.firstModLoad = false;
 				}
 			}
 			catch when (token.IsCancellationRequested) {
@@ -387,8 +_,8 @@
 			if (LastLaunchedTModLoaderVersion < version)
 				ShowWhatsNew = true;
 			*/
-			if (LastLaunchedTModLoaderVersion == new Version(0, 0))
+			//if (LastLaunchedTModLoaderVersion == new Version(0, 0))
-				ShowFirstLaunchWelcomeMessage = true;
+			//	ShowFirstLaunchWelcomeMessage = true;
 		}
 
 		/// <summary>
