--- src/tModLoader\Terraria.ModLoader\ModLoader.cs
+++ src/TerraCustom\Terraria.ModLoader\ModLoader.cs
@@ -122,7 +_,24 @@
 					OnSuccessfulLoad = null;
 				}
 				else {
-					Main.menuMode = 0;
+					//Main.menuMode = 0;
+					if (!Main.firstModLoad)
+					{
+						Main.menuMode = (int)TerraCustom.MenuModes.Settings;
+					}
+					else
+					{
+						Main.menuMode = (int)TerraCustom.MenuModes.ChooseWorldSize;
+					}
+					if (Mods.Length > 1)
+					{
+						Main.tModLoaderModsLoaded = true;
+					}
+					else
+					{
+						Main.tModLoaderModsLoaded = false;
+					}
+					Main.firstModLoad = false;
 				}
 			}
 			catch when (token.IsCancellationRequested) {

